File: xiaohashu-user-relation/xiaohashu-user-relation-biz/src/main/java/com/tongji/xiaohashu/user/relation/biz/constant/RedisKeyConstants.java
package com.tongji.xiaohashu.user.relation.biz.constant;

/** 
* @time 2025/3/27 10:19
* @author tongji
* @description 
*/
public class RedisKeyConstants {
    /**
     * 关注列表 KEY 前缀
     */
    private static final String USER_FOLLOWING_KEY_PREFIX = "following:";

    /**
     * 构建关注列表完整的 KEY
     */
    public static String buildUserFollowingKey(Long userId) {
        return USER_FOLLOWING_KEY_PREFIX + userId;
    }

    /**
     * 粉丝列表 KEY 前缀
     */
    private static final String USER_FANS_KEY_PREFIX = "fans:";

    /**
     * 构建粉丝列表完整的 KEY
     */
    public static String buildUserFansKey(Long userId) {
        return USER_FANS_KEY_PREFIX + userId;
    }
}

File: xiaohashu-count/xiaohashu-count-biz/src/main/java/com/tongji/xiaohashu/count/biz/constant/RedisKeyConstants.kt
package com.tongji.xiaohashu.count.biz.constant

/**
 * @time 2025/3/28 13:57
 * @author tongji
 * @description
 */
object RedisKeyConstants {
    /**
     * 用户维度计数 Key 前缀
     */
    private const val COUNT_USER_KEY_PREFIX = "count:user:"

    /**
     * Hash Field: 粉丝总数
     */
    const val FIELD_FANS_TOTAL = "fansTotal"

    /**
     * 构建用户维度计数 Key
     */
    @JvmStatic
    fun buildCountUserKey(userId: Long) = "$COUNT_USER_KEY_PREFIX$userId"

    /**
     * Hash Field: 关注总数
     */
    const val FIELD_FOLLOWING_TOTAL = "followingTotal"

    /**
     * 笔记维度计数 Key 前缀
     */
    private const val COUNT_NOTE_KEY_PREFIX = "count:note:"

    /**
     * Hash Field: 笔记点赞总数
     */
    const val FIELD_LIKE_TOTAL = "likeTotal"

    /**
     * 构建笔记维度计数 Key
     */
    @JvmStatic
    fun buildCountNoteKey(noteId: Long) = "$COUNT_NOTE_KEY_PREFIX$noteId"

    /**
     * Hash Field: 笔记收藏总数
     */
    const val FIELD_COLLECT_TOTAL = "collectTotal"

    /**
     * Hash Field: 笔记发布总数
     */
    const val FIELD_NOTE_TOTAL = "noteTotal"
}